src/libutil/error.cc

```diff
@@ -50,6 +50,40 @@ std::ostream & operator <<(std::ostream & str, const AbstractPos & pos)
     return str;
 }
 
+inline bool operator<(const AbstractPos& lhs, const AbstractPos& rhs)
+{
+    if (lhs.line != rhs.line)
+        return lhs.line < rhs.line;
+    if (lhs.column != rhs.column)
+        return lhs.column < rhs.column;
+    return false;
```

Comment:
What other cases can occur here other than line and column comparison? This makes the code and semantics harder to understand if the code leaves open what else can lead to the `false` return.